### PR TITLE
Refactor with TS

### PR DESCRIPTION
**Pull request by Eskibear**
See #19 

In this PR, I refactor the code with TS, and adopt async/await promise to handle callbacks.

### CODE REVIEW

**Comment by Eskibear** [comment]
As currently only NodeJS v8/v10/v12 are still under maintenance (Ref: [official docs](https://nodejs.org/en/about/releases/)), here I changed the .travis.yml accordingly.

File: package.json [hunk 1]
```diff
@@ -2,9 +2,12 @@
   "name": "find-java-home",
   "version": "0.2.0",
   "description": "Find JAVA_HOME on any system",
-  "main": "index.js",
+  "main": "out/index.js",
+  "types": "out/index.d.ts",
   "scripts": {
-    "test": "mocha -r should test.js"
+    "test": "mocha -r should test.js",
+    "prepack": "tsc -p .",
```
Comment (jsdevel) [inline package.json]: can you use prepare instead?

File: package.json [hunk 2]
```diff
@@ -2,9 +2,12 @@
   "name": "find-java-home",
   "version": "0.2.0",
   "description": "Find JAVA_HOME on any system",
-  "main": "index.js",
+  "main": "out/index.js",
```
Comment (jsdevel) [inline package.json]: can you write to `dist` instead?

File: .npmignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
+src/**
+**/*.map
```
Comment (jsdevel) [inline .npmignore]: maps should be published for debugging

File: .gitignore [hunk 4]
```diff
@@ -1 +1,3 @@
 /node_modules/
+.vscode
+out
```
Comment (jsdevel) [inline .gitignore]: dist instead of out

**Comment by Eskibear** [comment]
Comments addressed.

File: README.md [hunk 5]
```diff
@@ -17,6 +17,13 @@ require('find-java-home')(function(err, home){
   if(err)return console.log(err);
   console.log(home);
 });
+
+// OR
+import findJavaHome from ('find-java-home');
```
Comment (jsdevel) [inline README.md]: this can just be `from 'find-java-home';`

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    let res = undefined;
+    let err = null;
+    try {
+        res = await findJavaHomePromise(options);
+    } catch (error) {
+        err = error;
+    }
+    cb(err, res);
```
Comment (jsdevel) [inline src/index.ts]: a future improvement could be to allow `findJavaHome().then.catch` etc.

File: README.md (hunk 5 above)
Comment (Eskibear) [reply README.md]: copied from above, I’m changing it now

File: src/index.ts (hunk 6 above)
Comment (Eskibear) [reply src/index.ts]: That’s good. Here in this PR I simply keep the exposed api unchanged

**Comment by Eskibear** [comment]
updated.

**Comment by jsdevel** [comment]
huge update @Eskibear !!! thank you so much!

**Pull request merged**